### PR TITLE
Feature/range conversion

### PR DESCRIPTION
**Pull request by krider2010**
Sadly still necessary with half of the String/NSString APIs. Also very
useful when working in a mixed Objective-C/Swift code base.

This is based on code found on stack overflow, and credited as such. Test cases added. A little contrived, but the latter is a much reduced version of what I originally needed this solution for.

### CODE REVIEW

File: Sources/String+Extensions.swift [hunk 1]
```diff
@@ -259,4 +259,50 @@ extension String {
...
+     to NSString instances.
+     
+     Taken from:
+     http://stackoverflow.com/questions/25138339/nsrange-to-rangestring-index
+     
+     - parameter range: The Range instance to conver to an NSRange.
+     
+     - returns: The NSRange converted from the input. This will always succeed.
+     */
+    public func nSRangeFromRange(range : Range<String.Index>) -> NSRange {
```
Comment (hsoi) [inline Sources/String+Extensions.swift]: Let's make that `NSRangeFromRange()`. Yeah, strictly it breaks convention, but it's also convention to say `NSRange` not `nSRange`. I think when someone would be logically typing this out, they'd more likely type `NSRange` than `nSRange` (set aside code completion).

**Comment by hsoi** [comment]
Useful addition. Apart from the one naming thing, 👍

File: Sources/String+Extensions.swift (hunk 1 above)
Comment (krider2010) [inline Sources/String+Extensions.swift]: I was undecided about it anyways :)

**Comment by kreeger** [comment]
_Very_ useful. Looks good to me! 👍

**Pull request merged**